include/zpp/atomic_bitset.hpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2019 Erwin Rol <erwin@erwinrol.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR__INCLUDE__ZPP__ATOMIC_BITSET_HPP
+#define ZEPHYR__INCLUDE__ZPP__ATOMIC_BITSET_HPP
+
+#include <zpp/compiler.hpp>
+
+#include <atomic.h>
+#include <sys/__assert.h>
+
+#include <cstddef>
+
+namespace zpp {
```

Comment:
I think the namespace should be `zephyr`.  There's no reason to make it obscure.

Reply:
The reason for zpp is because it is short, trying to reduce the "need" to plaster *.cpp files with using namespace zephyr;

With your RAII I am not really following you. The zephyr "objects" are not "wrapped" the zpp::mutex _is_ the zephyr object, zpp::mutex m; does resource allocation and initialization (RAII). See also my comments on your comments below.

Reply:
It's also obscure.  Three extra characters to accurately represent the name of the project won't kill us.

Reply:
Ppl suggested that Zephyr didn't need any C API prefixing because the `z_` would be extra typing ;-) And `z::` I think is to short, and `zpp::` types real easy, just like `std::` 
I think 'zpp::' is nice and short and easy to remember. I don't really like the idea that all files than also land in the zephyr include dir, I think the C++ are better off in their own dir like they are now in the `include/zpp/` dir. Also `#include <zpp/mutex.hpp>` is shorter than `#include <zephyr/mutex.hpp>` 

Wouldn't mind hearing some second opinions from others on it, but it won't be a show stopper. I have the feeling our ideas on other parts are still miles apart though.